src/uu/cksum/src/cksum.rs

```diff
@@ -140,7 +140,20 @@ fn cksum(fname: &str) -> io::Result<(u32, usize)> {
     let mut rd: Box<dyn Read> = match fname {
         "-" => Box::new(stdin()),
         _ => {
-            file = File::open(&Path::new(fname))?;
+            let path = &Path::new(fname);
```

Comment:
could you please also add tests to cover these cases? thanks

Reply:
Ah yeah, i added them in #1923 but the tests i added found a small bug on freebsd and led me to do this.